bin/kvstore_v2/src/main.rs

```diff
@@ -0,0 +1,176 @@
+use api::{check_bootstrap_config, consensus_api::ConsensusEngine, NodeConfig};
+use api_types::{default_recover::DefaultRecovery, ConsensusApi, ExecutionChannel, ExecutionLayer};
+use clap::Parser;
+use cli::Cli;
+use execution_channel::ExecutionChannelImpl;
+use flexi_logger::{detailed_format, FileSpec, Logger, WriteMode};
+use pipeline_blockchain::*;
+use server::ServerApp;
+use std::{error::Error, sync::Arc, thread};
+
+struct TestConsensusLayer {
+    consensus_engine: Arc<dyn ConsensusApi>,
+}
+
+impl TestConsensusLayer {
+    fn new(node_config: NodeConfig, execution_client: Arc<dyn ExecutionChannel>) -> Self {
+        let execution_layer = ExecutionLayer {
+            execution_api: execution_client,
+            recovery_api: Arc::new(DefaultRecovery {}),
+        };
+        Self { consensus_engine: ConsensusEngine::init(node_config, execution_layer, 1337) }
+    }
+
+    async fn run(self) {
+        loop {
+            tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
+        }
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    let cli = Cli::parse();
+    let gcei_config = check_bootstrap_config(cli.gravity_node_config.node_config_path.clone());
+    let log_dir = cli.log_dir.clone();
+    let _handle = Logger::try_with_str("info")
+        .unwrap()
+        .log_to_file(FileSpec::default().directory(log_dir))
+        .write_mode(WriteMode::BufferAndFlush)
+        .format(detailed_format)
+        .start()
+        .unwrap();
+    let storage = Arc::new(SledStorage::new("blockchain_db")?);
+    let mut blockchain = Blockchain::new(storage.clone());
+
+    let (ordered_block_sender, ordered_block_receiver) = futures::channel::mpsc::channel(10);
+    blockchain.process_blocks_pipeline(ordered_block_receiver).await?;
+    let execution_channel = Arc::new(ExecutionChannelImpl::new(ordered_block_sender));
+
+    cli.run(move || {
+        tokio::spawn(async move {
+            let server = ServerApp::new(execution_channel.clone(), blockchain.state(), storage);
+            let _ = thread::spawn(move || {
+                let cl = TestConsensusLayer::new(gcei_config, execution_channel);
+                tokio::runtime::Runtime::new().unwrap().block_on(cl.run());
+            });
+            server.start("127.0.0.1:9006").await.unwrap();
+        })
+    })
+    .await;
+
+    Ok(())
+}
+
+mod tests {
+    use futures::{
+        channel::{mpsc, oneshot},
+        SinkExt, StreamExt,
+    };
+    use log::info;
+    use tokio::time::interval;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_blockchain_processing() -> Result<(), Box<dyn Error>> {
+        let _handle = Logger::try_with_str("info")
+            .unwrap()
+            .log_to_file(
+                FileSpec::default().directory("/home/jingyue/projects/gravity-sdk/kvstore_v2/logs"),
+            )
```

Comment:
hardcoded path

Reply:
Whoa! This is some garbage test code i forget to delete. Gonna remove them as soon as possible.